packages/sign_in_with_apple/lib/sign_in_with_apple.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+import './authorization_credential.dart';
+import './credential_state.dart';
+
+export './authorization_credential.dart'
+    show
+        AuthorizationCredential,
+        AuthorizationCredentialAppleID,
+        AuthorizationCredentialPassword;
+export './credential_state.dart' show CredentialState;
+export './sign_in_with_apple_button/sign_in_with_apple_button.dart'
+    show SignInWithAppleButton, SignInWithAppleButtonStyle, IconAlignment;
+
+class SignInWithApple {
+  static const MethodChannel _channel =
```

Comment:
```suggestion
  static const MethodChannel channel =
```

Might be nice to expose the channel so we can mock it in tests and assure that we call a method on that channel

Reply:
I know that you introduced that pattern to the app, but was there any guidance on this?

Thinking about this, really surprised how both the static channel on the `SignInWithApple` class and then the channel created in tests don't interfere with each other.

Reply:
You can technically create multiple channels for the same channel name.

They will just send the data then with the same channel name to the native side. 
By default they use a shared messenger, so you can create two instances of a channel, set a mock handler on the first one, send data on the other one and the first would receive the call.

I think depending on just the channel name is probably worse, as that should be even a bigger implementation detail.